gluten-core/src/main/scala/io/glutenproject/expression/WindowFunctionsBuilder.scala

```diff
@@ -19,21 +19,26 @@ package io.glutenproject.expression
 
 import io.glutenproject.expression.ConverterUtils.FunctionConfig
 import io.glutenproject.substrait.expression.ExpressionBuilder
-import org.apache.spark.sql.catalyst.expressions.{Rank, RowNumber, WindowFunction}
+import org.apache.spark.sql.catalyst.expressions.{CumeDist, DenseRank, PercentRank, Rank, RowNumber, WindowFunction}
 
 object WindowFunctionsBuilder {
   def create(args: java.lang.Object, windowFunc: WindowFunction): Long = {
     val functionMap = args.asInstanceOf[java.util.HashMap[String, java.lang.Long]]
-    val functionName = windowFunc match {
-      case rowNumber: RowNumber =>
-        ConverterUtils.makeFuncName(
-          ConverterUtils.ROW_NUMBER, Seq(rowNumber.dataType), FunctionConfig.OPT)
-      case rank: Rank =>
-        ConverterUtils.makeFuncName(
-          ConverterUtils.RANK, Seq(rank.dataType), FunctionConfig.OPT)
-      case other =>
-        throw new UnsupportedOperationException(s"not currently supported: $other.")
+    val substraitFunc = ExpressionMappings.window_functions_map.get(windowFunc.getClass)
+    if (substraitFunc.isDefined) {
+      val functionName = windowFunc match {
+        case _: RowNumber | _: Rank | _: DenseRank | _: PercentRank | _: CumeDist =>
```

Comment:
Why we need this match? `ExpressionMappings.window_functions_map.get(windowFunc.getClass)` should have checked this

Reply:
You are right, I deleted the match.